keystorage/hashicorp/src/testFixtures/java/tech/pegasys/signers/hashicorp/dsl/HashicorpNode.java

```diff
@@ -12,86 +12,293 @@
  */
 package tech.pegasys.signers.hashicorp.dsl;
 
+import static org.assertj.core.api.Assertions.assertThat;
+
 import tech.pegasys.signers.hashicorp.dsl.certificates.SelfSignedCertificate;
 
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.cert.CertificateEncodingException;
+import java.util.HashMap;
 import java.util.Map;
 import java.util.Optional;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
-import com.github.dockerjava.api.DockerClient;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.awaitility.Awaitility;
+import org.zeroturnaround.exec.ProcessExecutor;
+import org.zeroturnaround.exec.ProcessResult;
+import org.zeroturnaround.exec.stream.LogOutputStream;
+import org.zeroturnaround.exec.stream.slf4j.Slf4jStream;
 
 public class HashicorpNode {
+  private static final Logger LOG = LogManager.getLogger();
+
+  // these should be set by gradle test tasks or manually in IDE for individual test execution
+  private static String vaultInstallDir = System.getProperty("vaultInstallDir");
+  private static String vaultBinary = System.getProperty("vaultBinary", "vault");
+
+  private static final String VAULT_ROOT_PATH = "secret";
 
   private final Optional<SelfSignedCertificate> serverTlsCertificate;
-  private final DockerClient dockerClient;
-  private HashicorpVaultDocker hashicorpVaultDocker;
+  private final Path vaultPath;
+  private final Pattern portPattern = Pattern.compile(".*cluster address: \".*:(\\d+)\".*");
+  private final ObjectMapper objectMapper = new ObjectMapper();
 
-  private HashicorpNode(final DockerClient dockerClient) {
-    this(dockerClient, null);
-  }
+  private final String host = "127.0.0.1";
+  private final AtomicInteger port = new AtomicInteger(0);
 
-  private HashicorpNode(
-      final DockerClient dockerClient, final SelfSignedCertificate serverTlsCertificate) {
-    this.dockerClient = dockerClient;
-    this.serverTlsCertificate = Optional.ofNullable(serverTlsCertificate);
-  }
+  // initialized after start
+  private HashicorpVaultCommands hashicorpVaultCommands;
+  private Future<ProcessResult> serverFuture;
+  private String vaultToken;
+
+  /**
+   * Create a local Hashicorp Vault process and start it.
+   *
+   * <p>System property vaultInstallDir and vaultBinary should be set before calling this method.
+   *
+   * @param withTls if TLS should be enabled
+   * @return A started instance of HashicorpNode
+   */
+  public static HashicorpNode createAndStartHashicorp(final boolean withTls) {
+    if (vaultInstallDir == null) {
```

Comment:
How do these get set if that test is run through the IDE and not gradle? Should be skipping the test if these don't exist instead?

Reply:
this is the DSL (testFixtures) part. The actual value is set in gradle in acceptanceTest. IntelliJ automatically uses gradle to run individual test as well. If someone running acceptance test in an IDE where gradle is not utilized, the property may required to be set in run configuration.